developer/src/kmc-package/src/compiler/package-validation.ts

```diff
@@ -1,19 +1,31 @@
 import { KmpJsonFile, CompilerCallbacks } from '@keymanapp/common-types';
 import { CompilerMessages } from './messages.js';
 
+// const SLexicalModelExtension = '.model.js';
+
+// The keyboard ID SHOULD adhere to this pattern:
+const KEYBOARD_ID_PATTERN_PACKAGE = /^[a-z_][a-z0-9_]*\.(kps|kmp)$/;
```

Comment:
Technically, isn't this pattern for Package ID? (though keyboard ID maybe the same)

The `\.(kps|kmp)$` are package-related extensions.

Reply:
Yes, that's why it's KEYBOARD_ID_PATTERN_**PACKAGE** -- there will be others later for .kmn, .kmx, etc. See the model id patterns as a ref.